stuff/cron/client_course.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/python3
+
+'''Processing course messages.'''
+
+import argparse
+import logging
+import os
+import sys
+import string
+import random
+import MySQLdb
+import MySQLdb.cursors
+import pika
+
+
+sys.path.insert(
+    0,
+    os.path.join(
+        os.path.dirname(os.path.dirname(os.path.realpath(__file__))), "."))
+import lib.db   # pylint: disable=wrong-import-position
+import lib.logs  # pylint: disable=wrong-import-position
+
+
+def generate_code() -> str:
+    '''Generate an aleatory code'''
+    code_str = string.ascii_letters + string.digits
+    return ''.join(random.sample(code_str, 10))
```

Comment:
```suggestion
_CODE_ALPHABET = string.ascii_letters + string.digits

def generate_code() -> str:
    '''Generate a random code'''
    return ''.join(random.choices(_CODE_ALPHABET, 10))
```

la diferencia entre [`random.sample`](https://docs.python.org/3/library/random.html#random.sample) y [`random.choices`](https://docs.python.org/3/library/random.html#random.choices) es que el segundo es con reemplazo. esto es necesario para tener más códigos posibles.

también, quizás sería buena idea eliminar letras / números que se puedan confundir (i, l, 1, etc.). quitar minúsculas también ayudaría bastante. qué tal usar sólo los 20 caracteres que usan en https://en.wikipedia.org/wiki/Open_Location_Code#Specification ? así igual serían 20^10 posibles códigos que "deberían ser suficientes para todos".

puntos extra si se agregan uno o dos caracteres al final que sirvan de checksum para fácilmente distinguir códigos mal-copiados.

Reply:
concretamente, se puede usar una cadena de 9 caracteres seguido de un caracter generado por al [algorithmo de Luhn mod N]9https://en.wikipedia.org/wiki/Luhn_mod_N_algorithm) (donde N en este caso sería 20). esto aún permite generar 512e9 certificados. si necesitamos más en algún momento, podemos extender la cadena.

Reply:
pensándolo bien, me gusta más este algoritmo: https://metacpan.org/dist/Noid/view/noid#NOID-CHECK-DIGIT-ALGORITHM